generators/java/index.cjs

```diff
@@ -0,0 +1,212 @@
+    }
+  }
+
+  _initializing() {
+    return {
+      validateFromCli() {
+        this.checkInvocationFromCLI();
+      },
+      sayHello() {
+        if (!this.showHello()) return;
+        this.log(chalk.white('⬢ Welcome to the JHipster Java ⬢'));
+      },
+      loadRuntimeOptions() {
+        this.loadRuntimeOptions();
+      },
+      loadOptionsConstants() {
+        this.loadJavaOptionsConstants();
+      },
+    };
+  }
+
+  get initializing() {
+    if (this.delegateToBlueprint) return;
+    return this._initializing();
+  }
+
+  _prompting() {
+    return {
+      async showPrompts() {
+        if (this.shouldSkipPrompts()) return;
+        await this.prompt(
+          [
+            {
+              name: PACKAGE_NAME,
+              type: 'input',
+              validate: input => this._validatePackageName(input),
+              message: 'What is your default Java package name?',
+              default: () => this._getDefaultPackageName(),
+            },
+            {
+              name: BUILD_TOOL,
+              type: 'list',
+              choices: () => this.BUILD_TOOL_PROMPT_CHOICES,
+              message: 'What tool do you want to use to build backend?',
+              default: () => this.BUILD_TOOL_DEFAULT,
+            },
+            {
+              name: PRETTIER_JAVA_INDENT,
+              type: 'input',
+              message: 'What is the Java indentation?',
+              default: defaultConfig[PRETTIER_JAVA_INDENT],
+            },
+          ],
+          this.config
+        );
+      },
+    };
+  }
+
+  get prompting() {
+    if (this.delegateToBlueprint) return;
+    return this._prompting();
+  }
+
+  _configuring() {
+    return {
+      configure() {
+        this.configureProjectName();
+        this.configureInit();
+        this.configureJava();
+      },
+    };
+  }
+
+  get configuring() {
+    if (this.delegateToBlueprint) return;
+    return this._configuring();
+  }
+
+  _composing() {
+    return {
+      async compose() {
+        if (!this.shouldComposeModular()) return;
+        await this.composeWithJavaDependencies();
+      },
+    };
+  }
+
+  get composing() {
+    if (this.delegateToBlueprint) return;
+    return this._composing();
+  }
+
+  _loading() {
+    return {
+      loadConstants() {
+        this.loadProjectNameConstants();
+        this.loadInitConstants();
+        this.loadJavaConstants();
+      },
+      loadConfig() {
+        this.loadProjectNameConfig();
+        this.loadInitConfig();
+        this.loadJavaConfig();
+      },
+      loadDerivedConfig() {
+        this.loadDerivedProjectNameConfig();
+        this.loadDerivedInitConfig();
+        this.loadDerivedJavaConfig();
+      },
+    };
+  }
+
+  get loading() {
+    if (this.delegateToBlueprint) return;
+    return this._loading();
+  }
+
+  _writing() {
+    return {
+      async writeFiles() {
+        if (this.shouldSkipFiles()) return;
+        await this.writeFilesToDisk(files);
```

Comment:
Won't it be the standard?

Reply:
Yes, why?

Reply:
So to include in the superclass and avoid repeating